fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/DepositAccountDataValidator.java

```diff
@@ -582,6 +604,33 @@ private void validateDepositTermDeatilForUpdate(final JsonElement element, final
         }
     }
 
+    private void validateForInterestCalc(final JsonElement element, final DataValidatorBuilder baseDataValidator) {
+
+        Long principalAmount = this.fromApiJsonHelper.extractLongNamed(principalAmountParamName, element);
+        baseDataValidator.reset().parameter(principalAmountParamName).value(principalAmount).notNull();
```

Comment:
You dont need 2 of these validations. You can chain them together:
```
baseDataValidator.reset().parameter(principalAmountParamName).value(principalAmount).notNull().longGreaterThanZero();
```

Reply:
noted.